lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs

```diff
@@ -44,6 +44,11 @@ instance IsOwned RndState where
 instance GenChange RndState where
     genChange s = (error "GenChange RndState unimplemented", s)
 
+-- Unlike sequential derivation, we can't derive an order from the index only
+-- (they are randomly generated), nor anything else in the address itself.
+--
+-- Therefore, we'll simply consider that addresses using the random address
+-- derivation scheme won't be ordered in any particular order.
 instance CompareDiscovery RndState where
     compareDiscovery _ _ _ = EQ
```

Comment:
@rvl, @KtorZ: I was going to ask (before the PR got merged).

Can this ever be (accidentally) abused in a context where we expect an equality test to return `False`?

I wonder whether it would be more appropriate to have `compareDiscovery` return a `Maybe Ordering`?

Reply:
I say use the `Eq` class if you want to check whether two RndStates are equal.

Reply:
@jonathanknowles I am not sure to understand the question here? 
This class isn't about comparing `RndState` but about comparing the discovery date of addresses. In the context of random addresses, we simply consider there's no particular ordering and they're all discovered "at the same time".